subsys/sal/sid_pal/src/sid_ble_connection.c

```diff
@@ -75,6 +75,152 @@ static void ble_connect_cb(struct bt_conn *conn, uint8_t err)
+		return "SCO_INTERVAL_REJECTED";
+	case BT_HCI_ERR_SCO_AIR_MODE_REJECTED:
+		return "SCO_AIR_MODE_REJECTED";
+	case BT_HCI_ERR_INVALID_LL_PARAM:
+		return "INVALID_LL_PARAM";
+	case BT_HCI_ERR_UNSPECIFIED:
+		return "UNSPECIFIED";
+	case BT_HCI_ERR_UNSUPP_LL_PARAM_VAL:
+		return "UNSUPP_LL_PARAM_VAL";
+	case BT_HCI_ERR_ROLE_CHANGE_NOT_ALLOWED:
+		return "ROLE_CHANGE_NOT_ALLOWED";
+	case BT_HCI_ERR_LL_RESP_TIMEOUT:
+		return "LL_RESP_TIMEOUT";
+	case BT_HCI_ERR_LL_PROC_COLLISION:
+		return "LL_PROC_COLLISION";
+	case BT_HCI_ERR_LMP_PDU_NOT_ALLOWED:
+		return "LMP_PDU_NOT_ALLOWED";
+	case BT_HCI_ERR_ENC_MODE_NOT_ACCEPTABLE:
+		return "ENC_MODE_NOT_ACCEPTABLE";
+	case BT_HCI_ERR_LINK_KEY_CANNOT_BE_CHANGED:
+		return "LINK_KEY_CANNOT_BE_CHANGED";
+	case BT_HCI_ERR_REQUESTED_QOS_NOT_SUPPORTED:
+		return "REQUESTED_QOS_NOT_SUPPORTED";
+	case BT_HCI_ERR_INSTANT_PASSED:
+		return "INSTANT_PASSED";
+	case BT_HCI_ERR_PAIRING_NOT_SUPPORTED:
+		return "PAIRING_NOT_SUPPORTED";
+	case BT_HCI_ERR_DIFF_TRANS_COLLISION:
+		return "DIFF_TRANS_COLLISION";
+	case BT_HCI_ERR_QOS_UNACCEPTABLE_PARAM:
+		return "QOS_UNACCEPTABLE_PARAM";
+	case BT_HCI_ERR_QOS_REJECTED:
+		return "QOS_REJECTED";
+	case BT_HCI_ERR_CHAN_ASSESS_NOT_SUPPORTED:
+		return "CHAN_ASSESS_NOT_SUPPORTED";
+	case BT_HCI_ERR_INSUFF_SECURITY:
+		return "INSUFF_SECURITY";
+	case BT_HCI_ERR_PARAM_OUT_OF_MANDATORY_RANGE:
+		return "PARAM_OUT_OF_MANDATORY_RANGE";
+	case BT_HCI_ERR_ROLE_SWITCH_PENDING:
+		return "ROLE_SWITCH_PENDING";
+	case BT_HCI_ERR_RESERVED_SLOT_VIOLATION:
+		return "RESERVED_SLOT_VIOLATION";
+	case BT_HCI_ERR_ROLE_SWITCH_FAILED:
+		return "ROLE_SWITCH_FAILED";
+	case BT_HCI_ERR_EXT_INQ_RESP_TOO_LARGE:
+		return "EXT_INQ_RESP_TOO_LARGE";
+	case BT_HCI_ERR_SIMPLE_PAIR_NOT_SUPP_BY_HOST:
+		return "SIMPLE_PAIR_NOT_SUPP_BY_HOST";
+	case BT_HCI_ERR_HOST_BUSY_PAIRING:
+		return "HOST_BUSY_PAIRING";
+	case BT_HCI_ERR_CONN_REJECTED_DUE_TO_NO_CHAN:
+		return "CONN_REJECTED_DUE_TO_NO_CHAN";
+	case BT_HCI_ERR_CONTROLLER_BUSY:
+		return "CONTROLLER_BUSY";
+	case BT_HCI_ERR_UNACCEPT_CONN_PARAM:
+		return "UNACCEPT_CONN_PARAM";
+	case BT_HCI_ERR_ADV_TIMEOUT:
+		return "ADV_TIMEOUT";
+	case BT_HCI_ERR_TERM_DUE_TO_MIC_FAIL:
+		return "TERM_DUE_TO_MIC_FAIL";
+	case BT_HCI_ERR_CONN_FAIL_TO_ESTAB:
+		return "CONN_FAIL_TO_ESTAB";
+	case BT_HCI_ERR_MAC_CONN_FAILED:
+		return "MAC_CONN_FAILED";
+	case BT_HCI_ERR_CLOCK_ADJUST_REJECTED:
+		return "CLOCK_ADJUST_REJECTED";
+	case BT_HCI_ERR_SUBMAP_NOT_DEFINED:
+		return "SUBMAP_NOT_DEFINED";
+	case BT_HCI_ERR_UNKNOWN_ADV_IDENTIFIER:
+		return "UNKNOWN_ADV_IDENTIFIER";
+	case BT_HCI_ERR_LIMIT_REACHED:
+		return "LIMIT_REACHED";
+	case BT_HCI_ERR_OP_CANCELLED_BY_HOST:
+		return "OP_CANCELLED_BY_HOST";
+	case BT_HCI_ERR_PACKET_TOO_LONG:
+		return "PACKET_TOO_LONG";
+	case BT_HCI_ERR_TOO_LATE:
+		return "TOO_LATE";
+	case BT_HCI_ERR_TOO_EARLY:
+		return "TOO_EARLY";
+	default:
+		return "!!! UNEXPECTED HCI ERROR !!!";
+	}
+}
+
```

Comment:
I think it can be moved out to separated file, maybe to sidewalk/utils. It's a long code, and I think it can be reused in other sample